lib/basic_operations.py

```diff
@@ -7,15 +7,30 @@ class BasicOperations:
     @staticmethod
     def add(*args):
         """Adds any number of arguments and returns the result"""
-        for i in args:
-            if type(i) != int:
-                raise Exception
+        # check first how many arguments was passed
+        # if none, return zero
+        # if one, return that
+        # if more than one, get the type of the first one
+        # then use this to check the type of the other arguments
+        string = ""
         if len(args) == 0:
             return 0
         elif len(args) == 1:
             return args[0]
         else:
-            return sum(args)
+            if type(args[0]) == str:
+                for i in args:
+                    if type(i) != str:
```

Comment:
you don't have to check the type of the first element if it is a string because you already checked it at the if statement before this line. so to have one less loop iteration, you can write
`for i in args[1:]:`